tst/EnergyPlus/unit/WeatherManager.unit.cc

```diff
@@ -469,6 +511,91 @@ TEST_F(EnergyPlusFixture, WaterMainsCorrelationFromStatFileTest)
+    int totDays = std::accumulate(nDaysInMonth.begin(), nDaysInMonth.end(), 0);
+    double annualAvgOADryBulbTemp =
+        std::inner_product(std::begin(nDaysInMonth), std::end(nDaysInMonth), std::begin(monthlyDryBulbTemps), 0.0) / totDays;
+
+    const auto [min, max] = std::minmax_element(std::begin(monthlyDryBulbTemps), std::end(monthlyDryBulbTemps));
+    double monthlyAvgOADryBulbTempMaxDiff = (*max) - (*min);
+
+    EXPECT_TRUE(state->dataWeatherManager->OADryBulbAverage.OADryBulbWeatherDataProcessed);
+    EXPECT_NEAR(state->dataWeatherManager->OADryBulbAverage.AnnualAvgOADryBulbTemp, 9.988219178082193, 0.01);
+    EXPECT_NEAR(state->dataWeatherManager->OADryBulbAverage.MonthlyAvgOADryBulbTempMaxDiff, 28.7, 0.01);
+    EXPECT_NEAR(state->dataWeatherManager->OADryBulbAverage.AnnualAvgOADryBulbTemp, annualAvgOADryBulbTemp, 0.01);
+    EXPECT_NEAR(state->dataWeatherManager->OADryBulbAverage.MonthlyAvgOADryBulbTempMaxDiff, monthlyAvgOADryBulbTempMaxDiff, 0.01);
+
+    // January 15th water mains temperature test
+    state->dataEnvrn->DayOfYear = 15; // January 15th
+    WeatherManager::CalcWaterMainsTemp(*state);
+    EXPECT_NEAR(state->dataEnvrn->WaterMainsTemp, 7.5295, 0.0001);
+}
+
+TEST_F(EnergyPlusFixture, WaterMainsCorrelationFromStatFileTest_ActualBroken)
+{
+
+    state->files.inStatFilePath.filePath = configured_source_directory() / "tst/EnergyPlus/unit/Resources/broken.stat";
+
+    std::string const idf_objects = delimited_string({
+        "   Site:WaterMainsTemperature,",
+        "   CorrelationFromWeatherFile;  !- Calculation Method",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool foundErrors(false);
+    WeatherManager::GetWaterMainsTemperatures(*state, foundErrors);
+    EXPECT_FALSE(foundErrors); // expect no errors
+    EXPECT_TRUE(
+        compare_enums(state->dataWeatherManager->WaterMainsTempsMethod, WeatherManager::WaterMainsTempCalcMethod::CorrelationFromWeatherFile));
+    // for calculation method CorrelationFromWeatherFile these parameters are ignored
+    EXPECT_EQ(state->dataWeatherManager->WaterMainsTempsAnnualAvgAirTemp, 0.0);
+    EXPECT_EQ(state->dataWeatherManager->WaterMainsTempsMaxDiffAirTemp, 0.0);
+
+    EXPECT_TRUE(state->dataWeatherManager->WaterMainsParameterReport);
+
+    // CalcAnnualAndMonthlyDryBulbTemp was the one that was faulty
+    state->dataWeatherManager->OADryBulbAverage.CalcAnnualAndMonthlyDryBulbTemp(*state);
+    EXPECT_FALSE(state->dataWeatherManager->OADryBulbAverage.OADryBulbWeatherDataProcessed);
+    std::string const error_string = delimited_string({
+        "   ** Severe  ** CalcAnnualAndMonthlyDryBulbTemp: Stat file '" + state->files.inStatFilePath.filePath.string() +
+            "' does not have Monthly Statistics for Dry Bulb "
+            "temperatures.",
+        "   **   ~~~   ** Water Mains Temperature will be set to a fixed default value of 10.0 C.",
+    });
+
+    EXPECT_TRUE(compare_err_stream(error_string, true));
+}
```

Comment:
A test with a broken stat file, make sure the new severe is issued correctly.

Reply:
Nice.